GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs

```diff
@@ -340,6 +343,15 @@ void BindRepositories(IReadOnlyList<RecentRepoInfo> repos, bool isFavourite)
             }
         }
 
+        protected override void OnVisibleChanged(EventArgs e)
+        {
+            base.OnVisibleChanged(e);
+
+            // Reset the recent repo cache and the search
+            _controller.ClearCache();
```

Comment:
Why is this necessary?

Reply:
The intention was to clear the cache when the view is shown again to update the items shown in case something changed. This solution had a couple of flaws which I just resolved by clearing the cache in the `ShowRecentRepositories` (if required)